src/pip/_internal/operations/freeze.py

```diff
@@ -237,7 +238,7 @@ def get_requirement_info(dist):
 class FrozenRequirement(object):
     def __init__(self, name, req, editable, comments=()):
         # type: (str, Union[str, Requirement], bool, Iterable[str]) -> None
-        self.name = name
+        self.name = canonicalize_name(name)
```

Comment:
We should probably save this to e.g. `self.canonical_name` and leave name as-is so that it will still display properly in the pip freeze output.

Reply:
The reason I didn't do that is because the output of pip freeze uses the name property of the installation rather than the name of `FrozenRequirement`: https://github.com/pypa/pip/blob/f38b729ffbed0da90c5a5fc3969e6097124a9f7d/src/pip/_internal/operations/freeze.py#L170-L173

Reply:
It looks like `installations` is populated with `FrozenRequirement`s [here](https://github.com/pypa/pip/pull/6857/files#diff-b2eafa11b4e23079fbda84bb8103e702R82). It's good you point that block out though, since we would also be able to replace `canonicalize_name(installation.name)` there with `installation.canonical_name`.